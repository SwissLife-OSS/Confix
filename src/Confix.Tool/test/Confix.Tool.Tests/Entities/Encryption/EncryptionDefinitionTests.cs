using System.Text.Json.Nodes;
using Confix.Tool.Abstractions;
using FluentAssertions;
using Snapshooter.Xunit;

namespace Confix.Tool.Tests.Entities.Encryption;

public class EncryptionDefinitionTests
{
    [Fact]
    public void From_ProviderNull_ThrowsValidationException()
    {
        // Arrange
        var configuration = new EncryptionConfiguration(null);

        // Act
        var exception = Assert.Throws<ValidationException>(() => EncryptionDefinition.From(configuration));

        // Assert
        exception.Errors.Should().HaveCount(1);
    }

    [Fact]
    public void From_ValidProvider_ReturnsDefinition()
    {
        // Arrange
        var configuration = new EncryptionConfiguration(
            new EncryptionProviderConfiguration(
                "test", 
                null, 
                JsonNode.Parse("{}")!.AsObject()));

        // Act
        var definition = EncryptionDefinition.From(configuration);

        // Assert
        definition.MatchSnapshot();
    }
}
