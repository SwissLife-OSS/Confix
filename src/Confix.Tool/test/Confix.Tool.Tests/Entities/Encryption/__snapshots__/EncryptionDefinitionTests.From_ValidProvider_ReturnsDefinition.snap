{
  "Provider": {
    "Type": "test",
    "EnvironmentOverrides": {},
    "Value": {}
  }
}
